examples/_deployed/kitt/kitt.py

```diff
@@ -102,24 +102,42 @@ async def respond_to_image(user_msg: str):
             msg.images = []
 
         stream = await gpt.chat(initial_ctx)
-        await assistant.say(stream, allow_interruptions=True, add_to_ctx=True)
+        await assistant.say(stream, allow_interruptions=True)
+
+    async def get_human_video_track():
+        remote_video_tracks = [
+            t_pub.track
+            for _, p in ctx.room.participants.items()
+            for _, t_pub in p.tracks.items()
+            if t_pub.track is not None
+            and t_pub.kind == rtc.TrackKind.KIND_VIDEO
+            and isinstance(t_pub.track, rtc.RemoteVideoTrack)
+        ]
```

Comment:
I feel like this is harder to read than just having:
- a function called `_handle_video_track(track) ` that will create a VideoStream and iterate over the frame
- register a callback for track_subscribed and call this _handle_video_track when needed 
- iterate after the assistant start over all participant and their tracks to see if smthg already match

Reply:
Hrmm, well the strategy is the same but I like having both iteration and callback in one awaitable function because:
- it's self documenting: `get_human_video_track()` IMO is easier to understand than two different code paths (iteration & callback) that spawn a process_video_track task.
- it's easy to throw in a loop that ensures only one video track is being processed at a time
- in this case, it avoids having to create a new task and cancelling tasks if new video tracks come in

I've moved it out of main entrypoint for better readability (because really it's a utility function) and removed the process_video_track function and just inlined the video track at the end.

Reply:
no strong opinion here, either way is OK. I was more looking into this piece of code that is harder to read than a normal loop (at least for me)
![Screenshot 2024-06-07 at 11 24 11 PM](https://github.com/livekit/agents/assets/14841632/05b080a4-3361-4886-b4b0-47a8c8aa057e)

Reply:
ahh ok fair - updated to a loop

Reply:
i feel like having a list comprehension is more idiomatic for Python than callbacks, so i'm with Neil on this, if you need a tiebreaker :)

Reply:
too late haha. Anyways since agents is only on python for now, we might have enough non python folks reading it so 🤷‍♂️ We'll be making this kind of stuff easier soon enough anyways!